src/dbus/server/dbus_thread_object.cpp

```diff
@@ -1165,6 +1172,41 @@ otError DBusThreadObject::GetActiveDatasetTlvsHandler(DBusMessageIter &aIter)
     return error;
 }
 
+otError DBusThreadObject::SetFeatureFlagListDataHandler(DBusMessageIter &aIter)
+{
+#if OTBR_ENABLE_FEATURE_FLAGS
+    otError              error = OT_ERROR_NONE;
+    std::vector<uint8_t> data;
+    FeatureFlagList      featureFlagList;
+
+    VerifyOrExit(DBusMessageExtractFromVariant(&aIter, data) == OTBR_ERROR_NONE, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(featureFlagList.ParseFromString(std::string(data.begin(), data.end())), error = OT_ERROR_INVALID_ARGS);
```

Comment:
I guess the DBUS API caller supposed to pass in a serialized protobuf definition (string format?), right?
How can we ensure they are using the same protobuf definition?

Reply:
Yes (in byte format). For the upper layer we wraps the DBus interface and provides the Proto based API to thread_daemon so that developers can use it without mistake.